src/components/Swap/index.tsx

```diff
@@ -30,14 +31,15 @@ import useValidate from './useValidate'
 export interface SwapProps extends BrandingSettings, FeeOptions, SwapController, SwapEventHandlers, TokenDefaults {
   hideConnectionUI?: boolean
   routerUrl?: string
+  defaultChainId?: SupportedChainId
 }
 
 export default function Swap(props: SwapProps) {
   useValidate(props)
   useSyncController(props as SwapController)
   useSyncConvenienceFee(props as FeeOptions)
   useSyncSwapEventHandlers(props as SwapEventHandlers)
-  useSyncTokenDefaults(props as TokenDefaults)
+  useSyncTokenDefaults(props as TokenDefaults, props.defaultChainId)
```

Comment:
If `defaultChainId` is part of `useSyncTokenDefaults`, you should add it to `TokenDefaults` instead of passing it here separately.

This effectively means you can back out all changes to this file, and just make the change at the definition of the `TokenDefaults` interface.

Reply:
Will do! I also thought about it tbh when I will create the new param 😄